consensus/polybft/state.go

```diff
@@ -256,6 +335,79 @@ func (s *State) list() ([]*StateSyncEvent, error) {
 	return events, nil
 }
 
+// insertStateSyncEvent inserts a new state sync event to state event bucket in db
+func (s *State) insertExitEvent(event *ExitEvent) error {
+	return s.db.Update(func(tx *bolt.Tx) error {
+		raw, err := json.Marshal(event)
+		if err != nil {
+			return err
+		}
+
+		bucket := tx.Bucket(exitEventsBucket)
+
+		return bucket.Put(bytes.Join([][]byte{itob(event.Epoch), itob(event.ID), itob(event.BlockNumber)}, nil), raw)
+	})
+}
+
+// getExitEvent returns exit event with given id, which happened in given epoch and given block number
+func (s *State) getExitEvent(exitEventID, epoch, checkpointBlockNumber uint64) (*ExitEvent, error) {
+	var exitEvent *ExitEvent
+
+	err := s.db.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(exitEventsBucket)
+
+		key := bytes.Join([][]byte{itob(epoch), itob(exitEventID), itob(checkpointBlockNumber)}, nil)
+		v := bucket.Get(key)
+		if v == nil {
+			return &ExitEventNotFoundError{exitEventID, checkpointBlockNumber, epoch}
```

Comment:
You've switched `checkpointBlockNumber` and `epoch` fields initialization here.

I believe it is better to use field names when setting it, in order not to make permutations.

```suggestion
			return &ExitEventNotFoundError{
				exitID:          exitEventID,
				checkpointBlock: checkpointBlockNumber,
				epoch:           epoch,
			}
```

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/838/commits/a298d0ecad0757efc27415f6149f45b1f1e4dc7d